### PR TITLE
FISH-8465 Add support for remote local instance collection

### PR DESCRIPTION
**Pull request by kalinchan**
Created a new command to get the instance install root
Used this to pass to domain and log collector
You can add the `-p` parameter to the collect-diagnostics command to make heap dump, thread dump and jvm report work i.e. `./asadmin -p 24848 collect-diagnostics --target example`

I also moved all of the domain operations to a new DomainUtil class.

# Testing
Verified the following:

- Collecting Things™ from a local DAS
  - It should be possible to collect Things™ from a DAS that is local to where you run the asadmin command from:
    - If the DAS is online, All the Things™
    - If the DAS is offline, just the log and domain.xml
  - This includes accounting for scenarios where there may be more than one domain.
  - This includes accounting for scenarios where you're using a custom domain dir.
 
- Collecting Things™ from an instance on the same machine as the DAS
  - It should be possible to collect Things™ from an instance in a domain that is local to where you run the asadmin command from:
    - If the instance is online, All the Things™
    - If the instance is offline, just the log and domain.xml
  - This includes accounting for scenarios where there may be more than one domain.
  - This includes accounting for scenarios where you're using a custom node directory.
  - It should not matter if the DAS is running or not.

- Collecting Things™ from an instance that exists on a different machine to the DAS
  - It should be possible to collect Things™ from an instance in a domain that is local to where you run the asadmin command from:
    - If the instance is online, All the Things™
    - If the instance is offline, just the log and domain.xml
  - This includes accounting for scenarios where you're using a custom node directory.
  - It should not matter if the remote DAS is running or not.

# Notes
Previous PR should be merged first

### CODE REVIEW

File: src/main/java/fish/payara/extras/diagnostics/asadmin/CollectAsadmin.java [hunk 1]
```diff
@@ -172,162 +126,21 @@ private Map<String, Object> populateParameters(Map<String, Object> params) {
...
-                instances.forEach(s -> instanceXmlPaths.add(Paths.get(String.valueOf(nodePaths.get(nodeName)), s, "config", "domain.xml")));
-                continue;
-            }
-
-            if (pathType == PathType.LOG) {
-                instances.forEach(s -> instanceXmlPaths.add(Paths.get(String.valueOf(nodePaths.get(nodeName)), s, "logs")));
+        try {
+            if (this.getDomainsDir() != null) {
+                if (getDomainRootDir() != null) {
+                    params.put(ParamConstants.LOGS_PATH, getDomainRootDir().getPath() + "/logs");
```
Comment (Pandrex247) [inline src/main/java/fish/payara/extras/diagnostics/asadmin/CollectAsadmin.java:138]:
```suggestion
                    params.put(ParamConstants.LOGS_PATH, getDomainRootDir().getPath() + File.separator() + "logs");
```

File: src/main/java/fish/payara/extras/diagnostics/asadmin/LocalInstanceCommand.java
Comment (Pandrex247) [inline src/main/java/fish/payara/extras/diagnostics/asadmin/LocalInstanceCommand.java:1]: Why have you copied this file? You shouldn't need to

**Review by Pandrex247 — CHANGES_REQUESTED** [review]

File: src/main/java/fish/payara/extras/diagnostics/asadmin/LocalInstanceCommand.java
Comment (kalinchan) [reply src/main/java/fish/payara/extras/diagnostics/asadmin/LocalInstanceCommand.java:1]: I couldn't import it ):

Comment (Pandrex247) [reply src/main/java/fish/payara/extras/diagnostics/asadmin/LocalInstanceCommand.java:1]: Then add the dependency 😝
```
        <dependency>
            <groupId>fish.payara.server.internal.cluster</groupId>
            <artifactId>cluster-cli</artifactId>
            <version>${payara.version}</version>
        </dependency>
```

**Review by Pandrex247 — APPROVED** [review]

**Pull request merged**